thirdparty/litespeed-check.cls.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * Check if any plugins that could conflict with LiteSpeed Cache are active.
+ * @since		4.7
+ */
+
+namespace LiteSpeed\Thirdparty;
+
+defined( 'WPINC' ) || exit;
+
+class LiteSpeed_Check {
+
+	public static $_incompatible_plugins =
+		array(
+			'autoptimize/autoptimize.php',
+			'breeze/breeze.php',
+			'cache-enabler/cache-enabler.php',
+			'cachify/cachify.php',
+			'cloudflare/cloudflare.php',
+			'comet-cache/comet-cache.php',
+			'docket-cache/docket-cache.php',
+			'fast-velocity-minify/fvm.php',
+			'hummingbird-performance/wp-hummingbird.php',
+			'nginx-cache/nginx-cache.php',
+			'nitropack/main.php',
+			'pantheon-advanced-page-cache/pantheon-advanced-page-cache.php',
+			'powered-cache/powered-cache.php',
+			'sg-cachepress/sg-cachepress.php',
+			'simple-cache/simple-cache.php',
+			'redis-cache/redis-cache.php',
+			'w3-total-cache/w3-total-cache.php',
+			'wp-cloudflare-page-cache/wp-cloudflare-page-cache.php',
+			'wp-fastest-cache/wpFastestCache.php',
+			'wp-meteor/wp-meteor.php',
+			'wp-optimize/wp-optimize.php',
+			'wp-performance-score-booster/wp-performance-score-booster.php',
+			'wp-rocket/wp-rocket.php',
+			'wp-super-cache/wp-cache.php',
+		);
+
+	private static $_option = 'thirdparty_litespeed_check';
+	private static $_msg_id = 'id="lscwp-incompatible-plugin-notice"';
+
+	public static function detect() {
+		if ( ! is_admin() ) {
+			return;
+		}
+
+		/**
+		 * Check for incompatible plugins when `litespeed-cache` is first activated.
+		 */
+		$plugin = basename( LSCWP_DIR ) . '/litespeed-cache.php';
+		register_deactivation_hook( $plugin, function( $_network_wide ) {
+			\LiteSpeed\Admin_Display::delete_option( self::$_option );
+		} );
+		if ( ! \LiteSpeed\Admin_Display::get_option( self::$_option ) ) {
+			self::activated_plugin( $plugin, null );
+			\LiteSpeed\Admin_Display::add_option( self::$_option, true );
+		}
+
+		/**
+		 * Check for incompatible plugins when any plugin is (de)activated.
+		 */
+		add_action( 'activated_plugin', __CLASS__ . '::activated_plugin', 10, 2 );
+		add_action( 'deactivated_plugin', __CLASS__ . '::deactivated_plugin', 10, 2 );
+	}
+
+	public static function activated_plugin( $plugin, $network_wide ) {
+		self::incompatible_plugin_notice( $plugin, $network_wide, 'activated' );
```

Comment:
If the plugin is not in the incompatible list, may not need to call `incompatible_plugin_notice()`. Bcoz each call will cause 2 db operations right?

Reply:
It may also be good to show the msg each time when any plugin is installed

Reply:
> If the plugin is not in the incompatible list, may not need to call `incompatible_plugin_notice()`. Bcoz each call will cause 2 db operations right?

For future reference, it should be simple to add an early exit condition that detects whether the plugin that was just activated is in the incompatible plugins list. This check would be the first thing in the `incompatible_plugin_notice` function.